lib/takeaway.rb

```diff
@@ -0,0 +1,26 @@
+require 'total'
+require 'mainmenu'
+require 'deliverytime'
+class Order
+
+  attr_reader :items
+
+  def initialize(items = nil)
+    @items = items
+  end
+
+  def set_menu
+    @menu = MainMenu.new
```

Comment:
This is a 'hard-coded' dependency: the Order class will only be bug-free as long as the MainMenu class is also bug-free, so the classes are 'coupled'. This is something we want to avoid in Object-Oriented Design: the more that classes are dependent on each other, the more tangled and knotted the whole system will gradually become, until progress slows to a stop. The way to avoid this? _Dependency injection_. This is handled by the `initialize` method.

Once the dependency is injected (not hard-coded), then it becomes possible to test the classes independently too ('unit testing'), by using a double to stand in the place of MainMenu object. But as it is, line 13 is forced to use a real object, so there's no way to use a double. That means you are stuck when it comes to testing Order separately from testing MainMenu.